src_erl/NerlnetApp/src/Source/sourceSendingPolicies.erl

```diff
@@ -0,0 +1,106 @@
+  send_method_round_robin(TransmitterEts, Epochs, TimeInterval_ms, ClientWorkerPairs, BatchesListToSend, 0).
+send_method_round_robin(_TransmitterEts, Epochs, _TimeInterval_ms, _ClientWorkerPairs, _BatchesListToSend, EpochIdx) when EpochIdx == Epochs -> ok;
+send_method_round_robin(TransmitterEts, Epochs, TimeInterval_ms, ClientWorkerPairs, BatchesListToSend, EpochIdx) ->
+  % Sends a batch per each
+  ClientWorkerPairsIndexes = lists:seq(0, length(ClientWorkerPairs)-1),
+  ClientWorkerPairsWithIndexes = lists:zip(ClientWorkerPairsIndexes, ClientWorkerPairs), % Tuple {Idx, Triplet}
+  ClientWorkerPairsMap = maps:from_list(ClientWorkerPairsWithIndexes),
+  
+  BatchFunc = fun({{BatchIdx, WorkerIdx}, Batch}) ->
+    ClientWorkerPair = maps:get(WorkerIdx, ClientWorkerPairsMap),
+    prepare_and_send(TransmitterEts, TimeInterval_ms, Batch, BatchIdx, [ClientWorkerPair])
+  end, % end of BatchFunc
+  TotalNumOfBatches = length(BatchesListToSend),
+  BatchesIndexes = generate_batch_indexes(TotalNumOfBatches, EpochIdx),
+  ClientWorkerPairsLength = length(ClientWorkerPairs),
+  WorkerIdxBatchIdxTuples = [ {X , X rem ClientWorkerPairsLength} || X <- BatchesIndexes],
+  BatchesWithIndexes = lists:zip(WorkerIdxBatchIdxTuples, BatchesListToSend),
+  lists:foreach(BatchFunc, BatchesWithIndexes),
+  % update batches sent
+  SkippedBatches = ets:lookup_element(TransmitterEts, batches_skipped, ?DATA_IDX),
+  ets:update_counter(TransmitterEts, batches_sent, length(BatchesListToSend) - SkippedBatches),
+  send_method_round_robin(TransmitterEts, Epochs, TimeInterval_ms, ClientWorkerPairs, BatchesListToSend, EpochIdx + 1).
+
+
+send_method_random(TransmitterEts, Epochs, TimeInterval_ms, ClientWorkerPairs, BatchesListToSend) -> 
+  send_method_random(TransmitterEts, Epochs, TimeInterval_ms, ClientWorkerPairs, BatchesListToSend, 0).
+send_method_random(_TransmitterEts, Epochs, _TimeInterval_ms, _ClientWorkerPairs, _BatchesListToSend, EpochIdx) when EpochIdx == Epochs -> ok;
+send_method_random(TransmitterEts, Epochs, TimeInterval_ms, ClientWorkerPairs, BatchesListToSend, EpochIdx) ->
+  % Sends a batch per each
+  ClientWorkerPairsIndexes = lists:seq(1, length(ClientWorkerPairs)),
+  ClientWorkerPairsWithIndexes = lists:zip(ClientWorkerPairsIndexes, ClientWorkerPairs), % Tuple {Idx, Triplet}
+  ClientWorkerPairsMap = maps:from_list(ClientWorkerPairsWithIndexes),
+  BatchFunc = fun({{BatchIdx, WorkerIdx}, Batch}) ->
+    ClientWorkerPair = maps:get(WorkerIdx, ClientWorkerPairsMap),
+    prepare_and_send(TransmitterEts, TimeInterval_ms, Batch, BatchIdx, [ClientWorkerPair])
+  end, % end of BatchFunc
+  TotalNumOfBatches = length(BatchesListToSend),
+  BatchesIndexes = generate_batch_indexes(TotalNumOfBatches, EpochIdx),
+  ClientWorkerPairsLength = length(ClientWorkerPairs),
+  WorkerIdxBatchIdxTuples = [ {X , rand:uniform(ClientWorkerPairsLength)} || X <- BatchesIndexes],
```

Comment:
This is the optimization. 
I create list with pre-calculated random/rr indexes. 
This is better than calculate them on each function call since it reduces the number of operations performed between one sending process to another.